### PR TITLE
Fixed adding extra child just detach

### PR DESCRIPTION
**Pull request by matthew44-mappable**
Fixes #

### CODE REVIEW

File: src/controls/MMapGeolocationControl/index.ts [hunk 1]
```diff
@@ -131,6 +131,8 @@ class MMapGeolocationControl extends mappable.MMapGroupEntity<MMapGeolocationCon
     }
 
     protected override _onDetach() {
+        this._control.removeChild(this._button);
```
Comment (shmaxim-mappable) [inline src/controls/MMapGeolocationControl/index.ts:134]: don't forget to delete all entities created in `_onAttach`

or move its to constructor

File: src/controls/MMapSearchControl/index.ts [hunk 2]
```diff
@@ -252,6 +252,7 @@ class MMapSearchControl extends mappable.MMapComplexEntity<MMapSearchControlProp
     }
 
     protected override _onDetach(): void {
+        this._control.removeChild(this._search);
```
Comment (shmaxim-mappable) [inline src/controls/MMapSearchControl/index.ts:255]: Should delete `this._control` too

File: src/controls/MMapZoomControl/index.ts [hunk 3]
```diff
@@ -192,6 +192,7 @@ class MMapZoomControl extends mappable.MMapComplexEntity<MMapZoomControlProps, D
     }
 
     protected override _onDetach(): void {
+        this._control.removeChild(this._zoom);
```
Comment (shmaxim-mappable) [inline src/controls/MMapZoomControl/index.ts:195]: so you can delete only `this._control`, its subtree will be automatically detached

File: src/markers/MMapDefaultMarker/index.ts [hunk 4]
```diff
@@ -196,6 +196,10 @@ export class MMapDefaultMarker extends mappable.MMapComplexEntity<MMapDefaultMar
         this._marker.update({...this._props, onClick: this._onMarkerClick});
     }
 
+    protected override _onDetach(): void {
+        this.removeChild(this._marker);
```
Comment (shmaxim-mappable) [inline src/markers/MMapDefaultMarker/index.ts:200]: Should remove `this._popup` too

File: src/markers/MMapPopupMarker/index.ts [hunk 5]
```diff
@@ -138,6 +138,10 @@ export class MMapPopupMarker extends mappable.MMapComplexEntity<MMapPopupMarkerP
         this._marker.update(this._props);
     }
 
+    protected _onDetach(): void {
+        this.removeChild(this._marker);
```
Comment (shmaxim-mappable) [inline src/markers/MMapPopupMarker/index.ts:142]: same

**Review by shmaxim-mappable — CHANGES_REQUESTED** [review]

File: src/markers/MMapPopupMarker/index.ts (hunk 5 above)
Comment (matthew44-mappable) [reply src/markers/MMapPopupMarker/index.ts:142]: There is no `this.popup`, only `this.marker`, which is already being removed from children

**Comment by shmaxim-mappable** [comment]
lgtm

**Pull request merged**